src/authenticator.rs

```diff
@@ -1068,8 +1068,13 @@ enum StorageType {
 
 #[cfg(feature = "hyper-rustls")]
 fn install_crypto_provider_if_not_set() {
```

Comment:
instead of setting default provider when none set, should use [with_provider_and_native_roots](https://docs.rs/hyper-rustls/latest/hyper_rustls/struct.HttpsConnectorBuilder.html#method.with_provider_and_native_roots) in connector builder implementations

I can make a seperate PR that would use ring provider, & then this PR would need to fix that to use feature flags to decide

Reply:
Makes sense! 👌